src/endpoint/endpoint.js

```diff
@@ -231,18 +244,21 @@ function create_endpoint_handler(rpc, internal_rpc_client, options) {
         http_utils.parse_url_query(req);
 
         if (req.url.startsWith('/2015-03-31/functions')) {
-            req.func_sdk = new FuncSDK(rpc.new_client());
+            req.func_sdk = new FuncSDK(rpc_client);
             return lambda_rest_handler(req, res);
         }
 
         if (req.headers['x-ms-version']) {
-            req.object_sdk = new ObjectSDK(rpc.new_client(), internal_rpc_client, object_io);
+            req.object_sdk = new ObjectSDK(rpc_client, internal_rpc_client, object_io);
             return blob_rest_handler(req, res);
         }
 
         req.virtual_hosts = VIRTUAL_HOSTS;
-        req.object_sdk = new ObjectSDK(rpc.new_client(), internal_rpc_client, object_io);
-        return s3_rest_handler(req, res);
+        req.object_sdk = new ObjectSDK(rpc_client, internal_rpc_client, object_io);
+        const ret = await s3_rest_handler(req, res);
+
+        rpc_client_pool.release(rpc_client);
```

Comment:
What about calling release in all the other cases?

Reply:
Good catch. I didn't notice that we have early returns

Reply:
Changed to use a finally clause to catch all exit conditions (even throws)